src/objective/regression_obj.cu

```diff
@@ -83,8 +88,10 @@ class RegLossObj : public ObjFunction {
     // for better performance.
     const size_t n_data_blocks = std::max(static_cast<size_t>(1), (on_device ? ndata : nthreads));
     const size_t block_size = ndata / n_data_blocks + !!(ndata % n_data_blocks);
+    auto const n_targets = std::max(info.labels.Shape(1), static_cast<size_t>(1));
+
```

Comment:
We should use more ndarray friendly ways to express the calculation in the future.